src/main/java/org/elasticsearch/index/query/TermsFilterParser.java

```diff
@@ -181,15 +181,12 @@ public Filter parse(QueryParseContext parseContext) throws IOException, QueryPar
         }
 
         if (lookupId != null) {
-            // if there are no mappings, then nothing has been indexing yet against this shard, so we can return
-            // no match (but not cached!), since the Terms Lookup relies on the fact that there are mappings...
-            if (fieldMapper == null) {
-                return Queries.MATCH_NO_FILTER;
+            final TermsLookup lookup = new TermsLookup(lookupIndex, lookupType, lookupId, lookupRouting, lookupPath, parseContext);
+            final GetResponse getResponse = client.get(new GetRequest(lookup.getIndex(), lookup.getType(), lookup.getId()).preference("_local").routing(lookup.getRouting())).actionGet();
+            if (getResponse.isExists()) {
+                List<Object> values = XContentMapValues.extractRawValues(lookup.getPath(), getResponse.getSourceAsMap());
+                terms.addAll(values);
             }
```

Comment:
yeah it might be ok to do this since we are only parsing it on a search thread. I wonder if we can add an assertion for the threadpool that it is not a nework thread ever? but opening a diff issue is good